test/JDBC/expected/openquery_upgrd-vu-prepare.out

```diff
@@ -12,6 +12,112 @@ GO
 EXEC sp_addlinkedsrvlogin @rmtsrvname = 'bbf_server', @useself = 'FALSE', @rmtuser = 'jdbc_user', @rmtpassword = '12345678'
 GO
 
+-- Test sp_serveroption procedure
+EXEC sp_addlinkedserver  @server = N'bbf_server_1', @srvproduct=N'', @provider=N'SQLNCLI', @datasrc=N'localhost', @catalog=N'master'
+GO
+
+-- Add jdbc_user as linked server login
+EXEC sp_addlinkedsrvlogin @rmtsrvname = 'bbf_server_1', @useself = 'FALSE', @rmtuser = 'jdbc_user', @rmtpassword = '12345678'
+GO
+
+-- sp_serveroption with invalid server name. Should throw error
+EXEC sp_serveroption @server='invalid_server', @optname='query timeout', @optvalue='1'
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: server "invalid_server" does not exist)~~
+
+
+-- sp_serveroption with invalid server option. Should throw error
+EXEC sp_serveroption @server='bbf_server_1', @optname='invalid option', @optvalue='1'
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: Invalid option provided for sp_serveroption)~~
+
+
+-- sp_serveroption with server as NULL. Should throw error
+EXEC sp_serveroption @server=NULL, @optname='query timeout', @optvalue='1'
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: @server parameter cannot be NULL)~~
+
+
+-- sp_serveroption with optname as NULL. Should throw error
+EXEC sp_serveroption @server='bbf_server_1', @optname=NULL, @optvalue='1'
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: @optname parameter cannot be NULL)~~
+
+
+-- sp_serveroption with optvalue as NULL. Should throw error
+EXEC sp_serveroption @server='bbf_server_1', @optname='query timeout', @optvalue=NULL
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: @optvalue parameter cannot be NULL)~~
+
+
+-- sp_serveroption with negative optvalue
+EXEC sp_serveroption @server='bbf_server_1', @optname='query timeout', @optvalue='-5'
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: Invalid option value for query timeout.)~~
+
+
+-- sp_serveroption with float optvalue. should throw error
+EXEC sp_serveroption @server='bbf_server_1', @optname='query timeout', @optvalue='1.0001'
+GO
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: Invalid option value for query timeout.)~~
+
+
+-- sp_serveroption with optvalue greater than INT_MAX. should throw error
+EXEC sp_serveroption @server='bbf_server_1', @optname='query timeout', @optvalue='2147483650'
```

Comment:
Let's test with INT_MAX + 1 = 2147483648 and not 2147483650 to verify edge case

Reply:
done